### DISCUSSION TITLE
Help with logo AND text

### DISCUSSION THREAD

**Discussion by Paulo--M--** [chillerlan/php-qrcode#231] [Q&A]
Hello,
First of all, thanks for this wonderful piece of code.
While, following the examples, I can generate a QR code with a logo and a QR code with some text, I'm having trouble doing both: generating a QR code with **both** a logo and some text... I'm sure that I'm doing something silly and was wondering whether some kind soul could point me in the right direction. 
Many thanks in anticipation,
Paulo

**Comment by codemasher** [comment]
Hi! What output type are you planning to use? The approaches are very different for the various types.

**Comment by Paulo--M--** [comment]
Hello! Thank you for the quick reply :-)
For various reasons, I'm limited to creating a PNG.

**Reply by codemasher** [reply]
Oh that's great because there are 2 GD examples which you can combine: ["with text"](https://github.com/chillerlan/php-qrcode/blob/main/examples/imageWithText.php) and ["with logo"](https://github.com/chillerlan/php-qrcode/blob/main/examples/imageWithLogo.php) - please check them out to see if they suit your use case.

**Comment by Paulo--M--** [comment]
Hello again. Those are the two exact examples that I was referring to when I opened this thread! :-) :-) I can do both, separately, but for some odd reason I can't figure out how to put the two together... :-(  Any assistance would be much appreciated.

**Reply by codemasher** [reply]
Where exactly is the issue?

**Comment by Paulo--M--** [comment]
Well, it's probably down to my coding limitations but I'm having trouble combining the addition of the logo **and** the text, one after the other... Syntax problems come to mind :-) If you could provide an example to follow, that would be much appreciated. Thanks in advance.

**Reply by codemasher** [reply]
[For coding help I'd suggest StackOverflow or the PHPhelp reddit](https://php-qrcode.readthedocs.io/en/main/Appendix/Contribute.html#questions-and-issues) - when issues come up that are directly related to this library, you can check in. Thanks for understanding.

**Comment by Paulo--M--** [comment]
Just so that this isn't left without an answer... :-)  Since I couldn't figure out how to combine the two examples provided (my fault, no doubt), I ended up solving the problem using GD:
1. generate the PNG QR code
2. with GD, extend the canvas of the PNG
3. with GD, add text (with the font of choice, exactly where I wanted the text)
4. tested the resulting QR code -- success! :-) :-)

Thanks again for providing this great QR code generator.
